tsg/ranker/base.py

```diff
@@ -78,19 +119,29 @@ def cosine_score_calc(query_terms, index_dictionary_path=DICTIONARY_PATH,
 
     return scores
 
+def combine_and_or_scores(and_dict, or_dict):
+    sorted_and = sorted(and_dict.items(), key = operator.itemgetter(1,0))
+    sorted_or = sorted(or_dict.items(), key = operator.itemgetter(1, 0))
+
+    new_docs = set(sorted_or) - set(sorted_and)
+
+    scores = sorted_and + list(new_docs)
+
+    return scores
+
 def rank(query_terms, index_dictionary_path=DICTIONARY_PATH,
     index_info_path = INDEXINFO_PATH):
     '''
     Ranker takes a query and a dictionary path to calculates the score
     and retrieved a list of docs ordered by score and doc_id as
     tuples [(doc_1,score_1),(doc_2, score_2), ..., (doc_n,score_n)]
     '''
```

Comment:
And the idea was not, that we do conjunctive and disjunctive and then just combine them...
some terms may be conjunctive, some disjunctive.